CLI command neighbor X weight y

xschily opened the pull request:
Tags: fix, dev, Github issue #15

Change-Id: I233412563a5010966d99ef008a2b73d9b8da3d07
Signed-off-by: Ilya Schepin ischepin@mera.ru

Implement BGP commands

`neighbor X weight <1-65535>`
`no neighbor X weight`
`no neighbor X weight <1-65535>`

Depends on MERAprojects/ops-cli#7

xschily committed:
CLI command neighbor X weight y

Tags: fix, dev, Github issue #15

Change-Id: I233412563a5010966d99ef008a2b73d9b8da3d07
Signed-off-by: Ilya Schepin <ischepin@mera.ru>

xschily committed:
CLI command neighbor X weight Y

Tags: fix, dev, Github issue #15

Change-Id: Ia6a81e42b936d09f3dd54ff7867b4d250d355880
Signed-off-by: Ilya Schepin <ischepin@mera.ru>

stroykov commented on bgpd/bgp_ovsdb_if.c:
```diff
@@ -2122,10 +2122,6 @@ void bgp_daemon_ovsdb_neighbor_update (struct peer *peer,
      */
 
-    VLOG_DBG("updating weight to %d\n", peer->weight);
-    ovsrec_bgp_neighbor_set_weight(ovs_bgp_neighbor_ptr,
```
your solution does not look like the legacy quagga one

xschily replied:
Which part exactly? This one was removed because BGP neighbor weight is a configuration parameter and it is not distributed to BGP neighbor. Basically it has only local meaning and only when a best route must be selected from BGP route table. There is no reason for BGPD to update this parameter upstream in OVSDB, it should only be changed by user and propagated to BGPD so it could assign weights to routes from different neighbors.

vkarpov commented:
No comments.

xschily commented on bgpd/bgp_ovsdb_if.c:
```diff
@@ -2410,6 +2406,21 @@ bgp_nbr_update_source_ovsdb_apply_changes (const struct ovsrec_bgp_neighbor *ovs
+    uint16_t weight;
+
+    if (COL_CHANGED (ovs_nbr, ovsrec_bgp_neighbor_col_weight, idl_seqno)) {
+        weight = (ovs_nbr->n_weight == 0 ? BGP_ATTR_DEFAULT_WEIGHT :
```
I suggest to avoid using function `peer_weight_unset` since CLI description of command `no neighbor weight` (written by ops designers) explicitly says ""Set default weight for routes from this neighbor", thus, it should not be reset to 0.

apolovin commented on bgpd/bgp_backend_functions.c:
```diff
@@ -2265,6 +2265,22 @@ daemon_neighbor_aspath_filter_cmd_execute(struct bgp *bgp, char *peer_str,
+
+  VLOG_DBG("Updating peer %s weight to %d", peer->host, weight);
+  peer_weight_set (peer, weight);
+}
```
no return!

apolovin commented on bgpd/bgp_ovsdb_if.c:
```diff
@@ -2410,6 +2406,21 @@ bgp_nbr_update_source_ovsdb_apply_changes (const struct ovsrec_bgp_neighbor *ovs
+    if (COL_CHANGED (ovs_nbr, ovsrec_bgp_neighbor_col_weight, idl_seqno)) {
+        weight = (ovs_nbr->n_weight == 0 ? BGP_ATTR_DEFAULT_WEIGHT :
+            (uint16_t) *ovs_nbr->weight);
+        daemon_neighbor_weight_cmd_execute (bgp_instance, name, weight);
```
if you do not check return value here why the function is non-void?

stroykov replied:
common function template used. No any intention here :)

apolovin reviewed (changes requested):
daemon_neighbor_weight_cmd_execute: at least add return statement or make it void

stroykov committed:
Merge branch 'master' into neighbor_weight_cmd

stroykov committed:
CLI command neighbor X weight y

Tags: fix, dev, Github issue #15

Change-Id: I233412563a5010966d99ef008a2b73d9b8da3d07
Signed-off-by: Ilya Schepin ischepin@mera.ru

Implement BGP commands

neighbor X weight <1-65535>
no neighbor X weight
no neighbor X weight <1-65535>

Depends on MERAprojects/ops-cli#7

stroykov commented:
https://review.openswitch.net/#/c/13892/

vkarpov commented:
Merged into meraswitch/devel/master.
